### PR TITLE
refactor header.js

### PR DESCRIPTION
**Pull request by lahirue**

### CODE REVIEW

File: modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/helpers/header.js [hunk 1]
```diff
@@ -17,9 +17,42 @@
...
+    o.css.push('sidepanel.css');
+    o.css.push('left-navigation.css');
+    o.css.push('font-wso2.css');
+    o.css.push('bootstrap-dialog.min.css');
+    o.css.push('typeahead.css');
+    o.js.push('sidepanel.js');
+    o.js.push('greg-subscriptions-api.js');
+    o.js.push('bootstrap-dialog.min.js');
+    o.js.push('typeahead.bundle.min.js');
+    // remove this line, because this file is loaded twice in store side. That occurs search-box css issue
```
Comment (sameerak) [inline modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/helpers/header.js]: remove commented out code

Comment (lahirue) [inline modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/helpers/header.js]: Fixed with this PR 
https://github.com/wso2/product-greg/pull/847

File: modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/helpers/header.js [hunk 2]
```diff
@@ -17,9 +17,42 @@
...
+var fn = that.resources||function() { return {} };
+var resources = function(page, meta) {
+    var o = fn(page, meta);
+    if (!o.css) {
+        o.css = [];
+    }
+    if(!o.js){
+        o.js = [];
+    }
+    if(!o.code){
```
Comment (splinter) [inline modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/helpers/header.js]: Code Formatting issue

File: modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/helpers/header.js [hunk 3]
```diff
@@ -17,9 +17,42 @@
...
+        that[name] = hps[name];
+    }
+}
+var fn = that.resources||function() { return {} };
+var resources = function(page, meta) {
+    var o = fn(page, meta);
+    if (!o.css) {
+        o.css = [];
+    }
+    if(!o.js){
```
Comment (splinter) [inline modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/helpers/header.js]: Code Formatting issue

File: modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/helpers/header.js (hunk 2 above)
Comment (lahirue) [inline modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/helpers/header.js]: Thank you Sameera for mentioning it,
This has been fixed with https://github.com/wso2/carbon-store/pull/834

File: modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/helpers/header.js (hunk 3 above)
Comment (lahirue) [inline modules/es-extensions/store/extensions/app/greg-store-defaults/themes/store/helpers/header.js]: Thank you Sameera for mentioning it,
This has been fixed with wso2/carbon-store#834

**Pull request merged**